src/legacy/ui/public/field_editor/field_editor.js

```diff
@@ -315,9 +317,14 @@ export class FieldEditorComponent extends PureComponent {
   renderType() {
     const { field, fieldTypes } = this.state;
     const { intl } = this.props;
+    const hasTypeConflict = field.conflictDescriptions && typeof field.conflictDescriptions === 'object';
 
     return (
-      <EuiFormRow label={intl.formatMessage({ id: 'common.ui.fieldEditor.typeLabel', defaultMessage: 'Type' })}>
+      <EuiFormRow
+        label={intl.formatMessage({ id: 'common.ui.fieldEditor.typeLabel', defaultMessage: 'Type' })}
+        helpText={hasTypeConflict && this.renderTypeConflict(field.conflictDescriptions)}
```

Comment:
Move `renderTypeConflict()`  to render after the `EuiSelect`, not inside the `helpText` which shouldn't contain more than plain text or some form of simple plain text.